tools/loader/Loader.cpp

```diff
@@ -240,6 +240,14 @@ void Loader::compile(Context &ctx) {
     // perform CSE, etc.
     ::optimize(F_, glow::CompilationMode::Infer);
 
+    // Lower everything, keeping track of what NodeValues were lowered to other
+    // NodeValues via the loweredMap_. This loweredMap_ is passed to
+    // generateNodeQuantizationInfos() when writing out the profile, allowing
+    // for both lowered and unlowered NodeValues to find their quantization
+    // parameters.
+    ::lower(F_, *EE_.getBackend(), &loweredMap_);
+    ::DCE(F_);
```

Comment:
Why is DCE not just the last step of `glow::lower()`? Is there any time we don't want to DCE after lowering?

Reply:
We don't always optimize for example in the onnxifi check compatibility case we don't. If the overhead of DCE is small-ish I think we should just automatically DCE after lowering to avoid leaking calls to DCE from places outside of optimizers like here.

Reply:
> We don't always optimize for example in the onnxifi check compatibility case we don't.

that's a bug :) You keep both lowered and prior lowering nodes for compatibility validation, which we should not do.

Reply:
@rdzhabarov yep it is, I think we should add DCE after lowering to fix it

Reply:
@jackm321 I think that adding `DCE` at the end of `lower()` could be reasonable. I think in practice DCE is pretty quick. Do you want to put up an issue or PR for that so we can discuss further?

Reply:
@jfix71 will do